launch/docker.go

```diff
@@ -94,19 +99,41 @@ func (d *docker) runBuild(buildEntry buildEntry) error {
 	artVol := fmt.Sprintf("%s/:%s", hostArtDir, containerArtDir)
 	binVol := fmt.Sprintf("%s:%s", d.volume, "/opt/sd")
 	habVol := fmt.Sprintf("%s:%s", d.habVolume, "/opt/sd/hab")
+
+	// Overwrite steps for sd-local interact mode. The env will load later.
+	if d.interactMode {
+		buildEntry.Steps = []screwdriver.Step{
+			{
+				Name:    "sd-local-init",
+				Command: "env > /tmp/sd-local.env",
+			},
+		}
+	}
+
 	configJSON, err := json.Marshal(buildEntry)
 	if err != nil {
 		return err
 	}
 
 	logrus.Infof("Pulling docker image from %s...", buildImage)
-	err = d.execDockerCommand("pull", buildImage)
+	_, err = d.execDockerCommand("pull", buildImage)
 	if err != nil {
 		return fmt.Errorf("failed to pull user image %v", err)
 	}
 
 	dockerCommandArgs := []string{"container", "run"}
-	dockerCommandOptions := []string{"--rm", "-v", srcVol, "-v", artVol, "-v", binVol, "-v", habVol, buildImage, "/opt/sd/local_run.sh", string(configJSON), buildEntry.JobName, environment["SD_API_URL"], environment["SD_STORE_URL"], logfilePath}
+	dockerCommandOptions := []string{"--rm", "-v", srcVol, "-v", artVol, "-v", binVol, "-v", habVol, buildImage}
+	configJSONArg := string(configJSON)
+	if d.interactMode {
+		configJSONArg = fmt.Sprintf("'%s'", configJSONArg)
```

Comment:
I think this work can be done all the time, not just in interactMode.

Reply:
When running with exec.Command, extra quotations can break user-defined steps if they are enclosed in extra quotes. Somewhere in the escape became incorrect, it waiting for the closing quotation to be typed. It's better to do nothing and let exec.Command do its job.
https://github.com/screwdriver-cd/sd-local/pull/45/files#diff-4f249559a0063721b9902972b3edda755ed28a3110c463e1de09ddb8bd87ecedL138

Reply:
I got it.